wordpress-mobile/WordPress-Android
# [Pages] Show virtual Homepage for block-based themes

Part of #18371

This implements the logic of showing the virtual Homepage item in the Pages List for sites that have a Block-based theme. The idea is to open the Site Editor on a WebView for those sites since the app does not support editing theme templates. The regular "legacy" home Page will be hidden if it exists.

The feature is currently hidden behind the `SiteEditorMVPFeatureConfig` feature flag (in development).

This is a partial PR, containing a few TODOs that will be resolved either:
- In a future PR (i.e.: callbacks implementation to open external links and the WebView)

Example of the virtual homepage item
![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/f32a2f21-fa66-4b8f-a921-26a30fddecff)

![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/ef773960-5528-4eb9-bf34-257a71777c80)

## To test
Requirements:
- Default case: At least 1 site using a block-based theme, such as `Vetro`, with the homepage set to display "your latest posts" (on the web you can find it at `Appearance > Customize > Homepage Settings`
- Special case: At least 1 site using a block-based theme, such as `Vetro`, with the homepage set to display "a static page" (on the web you can find it at `Appearance > Customize > Homepage Settings`
- At least 1 site using a non-block-based theme (or classic theme), such as `Confit`

### Testing Feature Flag off
This is just to ensure the feature is not active when the feature flag is off.
1. Login with your account
2. Select a site with a block-based theme
3. Go to `Debug Settings`
4. **Verify** the `SiteEditorMVPFeatureConfig` is OFF / not set
5. Go back to the `My Site` screen
6. Tap on `Pages`
7. **Verify** all pages show normally (there is no "virtual" homepage item)
8. Go back and select a site with a classic theme
9. Go back to the `My Site` screen
10. Tap on `Pages`
11. **Verify** all pages show normally (there is no "virtual" homepage item)

### Testing Feature flag on
1. Login with your account
2. Select a site with a block-based theme
3. Go to `Debug Settings`
4. Enable the `SiteEditorMVPFeatureConfig`
5. Go back to the `My Site` screen
6. Tap on `Pages`
7. **Verify** the "virtual" homepage item is shown
8. Tap the card
9. **Verify** it shows a snackbar saying `OPEN_SITE_EDITOR` 
10. Tap the info icon
11. **Verify** it shows a snackbar saying `OPEN_LEARN_MORE_URL`
12. Go back and select a site with a classic theme
13. Go back to the `My Site` screen
14. Tap on `Pages`
15. **Verify** the "virtual" homepage item is NOT shown

### Special case vs Default case
There's a special scenario where the site might be set to use a static page even though it is using a block-based theme. In that case, the static page set as "Home" will be replaced by the virtual homepage item. Just test the steps above with a site in this special case (see requirements).

## Regression Notes
1. Potential unintended areas of impact
Behavior issues in the Sites Lists in general.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual Testing.

3. What automated tests I added (or what prevented me from doing so)
Added testing checking behavior depending on feature flag and usage of block-based themes.

PR submission checklist:


UI Changes testing checklist:
